### DISCUSSION TITLE
vulnerabilities still returned after running pnpm audit --fix && pnpm install

### DISCUSSION THREAD

**Discussion by dagda1** [pnpm/pnpm#4333] [Q&A]
I first of all ran `pnpm audit --fix` in  a monorepo which created  the following enteries in the root package.json:

```json
  "pnpm": {
    "overrides": {
      "ansi-regex@>2.1.1 <5.0.1": ">=5.0.1",
      "nth-check@<2.0.1": ">=2.0.1",
      "immer@<9.0.6": ">=9.0.6",
      "tar@<4.4.18": ">=4.4.18",
      "browserslist@>=4.0.0 <4.16.5": ">=4.16.5",
      "trim@<0.0.3": ">=0.0.3",
      "postcss@<8.2.13": ">=8.2.13",
      "node-forge@<1.0.0": ">=1.0.0",
      "marked@<4.0.10": ">=4.0.10",
      "node-fetch@<2.6.7": ">=2.6.7",
      "glob-parent@<5.1.2": ">=5.1.2",
      "follow-redirects@<1.14.8": ">=1.14.8"
    }
  }
```

I then ran `pnpm install` followed by `pnpm audit`

which still returned entries for `follow-redirects` and `node-fetch` even though they are in the `overrides` section above.

is this because there needs to be `overrides` in other package.json files in the monorepo?

**Comment by zkochan** [comment]
In a workspace, the overrides are only needed in the root `package.json`.

**Comment by dagda1** [comment]
@zkochan running `pnpm why follow-redirects` is bringing back the correct version that fixed the vulnerability `1.14.8` so I think that is fine.  There was an entry in the `pnpm-lock.yaml` that pointed to `1.14.6`:

```yaml

  /http-proxy/1.18.1:
    resolution: {integrity: sha512-7mz/721AbnJwIVbnaSv1Cz3Am0ZLT/UBwkC92VlxhXv/k/BBQfM2fXElQNC27BVGr0uwUpplYPQM9LnaBMR5NQ==}
    engines: {node: '>=8.0.0'}
    dependencies:
      eventemitter3: 4.0.7
      follow-redirects: 1.14.6
      requires-port: 1.0.0
    transitivePeerDependencies:
      - debug
```

I updated the entry manually and it was removed from the list returned by `pnpm audit`.

Should adding the `overrides` and running `pnpm install` updated all `follow-redirects` to `1.14.8`?

**Reply by zkochan** [reply]
`1.14.6` is matched by `follow-redirects@<1.14.8`, so yes, it should be updated with those overrides.

**Comment by dagda1** [comment]
@zkochan thank you for the reply, do you think it is a bug?

**Reply by zkochan** [reply]
if it doesn't work automatically, then yes

**Comment by rfestag** [comment]
I am seeing this issue as well. On my end, `follow-redirects` is using version `1.14.7_debug@4.3.3`. Perhaps that's the reason it wasn't matching? In any case, for me the parent dependency was axios, which has been updated. So I added an entry for `"axios@<0.26.0": ">=0.26.0"` to my overrides instead.

**Comment by prabhatthapa** [comment]
Any update on this? I can still see this issue.

**Reply by weyert** [reply]
What’s the related issue ticket?

**Reply by prabhatthapa** [reply]
It looks like running `pnpm audit --fix` and then running `pnpm audit` is still complaining about the vulnerabilities

**Reply by prabhatthapa** [reply]
I had to ignore those vulnerabilities for now.

**Reply by weyert** [reply]
Did you raise an issue? Discussing it here doesn't mean it will be acted on

**Reply by mcm-ham** [reply]
https://github.com/pnpm/pnpm/issues/5943
